src/assets/wise5/classroomMonitor/dataExport/strategies/AbstractComponentDataExportStrategy.ts

```diff
@@ -0,0 +1,367 @@
+import { Component } from '../../../common/Component';
+import { ComponentDataExportParams } from '../ComponentDataExportParams';
+import { ComponentRevisionCounter } from '../ComponentRevisionCounter';
+import { AbstractDataExportStrategy } from './AbstractDataExportStrategy';
+
+export abstract class AbstractComponentDataExportStrategy extends AbstractDataExportStrategy {
+  canViewStudentNames: boolean;
+  columnNames: string[] = [];
+  columnNameToNumber: Map<string, number> = new Map<string, number>();
+  includeOnlySubmits: boolean;
+  includeStudentNames: boolean;
+  workSelectionType: string;
+
+  constructor(protected component: Component, additionalParams: ComponentDataExportParams) {
+    super();
+    this.canViewStudentNames = additionalParams.canViewStudentNames;
+    this.includeOnlySubmits = additionalParams.includeOnlySubmits;
+    this.includeStudentNames = additionalParams.includeStudentNames;
+    this.workSelectionType = additionalParams.workSelectionType;
+  }
+
+  abstract generateComponentWorkRow(
+    rowCounter: number,
+    workgroupId: number,
+    userId1: number,
+    userId2: number,
+    userId3: number,
+    studentName1: string,
+    studentName2: string,
+    studentName3: string,
+    periodName: string,
+    componentRevisionCounter: ComponentRevisionCounter,
+    componentState: any
+  ): string[];
+
+  generateComponentHeaderRow(
+    columnNames: string[],
+    columnNameToNumber: Map<string, number>
+  ): string[] {
+    this.populateColumnNames(columnNames, columnNameToNumber);
+    const headerRow = [];
+    for (const columnName of columnNames) {
+      headerRow.push(columnName);
+    }
+    return headerRow;
```

Comment:
Write this in one line for brevity (1 line vs 5)?
```suggestion
    return columnNames.map((columnName) => columnName);
    or 
    return [...columnNames];
```